### DISCUSSION TITLE
`useCopyPaste` hook prevents copy paste in general

### DISCUSSION THREAD

**Discussion by vedovelli** [xyflow/xyflow#4117] [General]
Hello all! 👋

I'm using the PRO example and my struggle is to confine the copy and paste handlers to the workflow itself.

My focus is in this block of code:

```typescript
  // initialize the copy/paste hook
  // 1. remove native copy/paste/cut handlers
  // 2. add mouse move handler to keep track of the current mouse position
  useEffect(() => {
    const events = ["cut", "copy", "paste"];

    if (rfDomNode) {
      const preventDefault = (e: Event) => {
        console.log("🔥", e);
        e.preventDefault();
      };
```

The `console.log` was included by me and my intention is to only prevent default if target is the workflow.

It turns out it is never called.

Any help is much appreciated!

**Comment by moklick** [comment]
Hey @vedovelli 

thanks for the report! The "copy" event added via `addEventListener` gets called when you use the interface of your browser (Edit -> copy). Unfortunately our `useKeyPress` hook doesn't give you access to the actual event. Maybe you could check if the React Flow DOM node is focused as another condition that needs to be true in order to call the callback of `useShortcut`?

**Comment by vedovelli** [comment]
Dear @moklick . Thanks for you answer. It prompted me to check the source code for `useKeyPress` and I've noticed it takes a `target` within `options`.

By passing `rfDomNode` as the target, it solved my issue! 🎉

![image](https://github.com/xyflow/xyflow/assets/151001/646c3b47-b183-450b-895a-3d090d7c4347)


Once again, thanks a bunch.

Vedovelli

**Reply by moklick** [reply]
I totally forgot about that option 👌

**Reply by vedovelli** [reply]
Would you consider updating the README.md for the example, so future users might benefit from it?

**Reply by moklick** [reply]
Yep! We will check it.

**Comment by addeelnayyer** [comment]
@moklick I am using useCopyPaste hook, I have some custom nodes which have text inputs, the issue is I am passing rfDomNode as target but I am unable to paste simple text copied on clipboard in the custom node

**Comment by lidiaCirrone** [comment]
Same as @addeelnayyer  :( any suggestions anyone?
